### PR TITLE
Domains: Added transfer start functionality for domains in pending_start state.

### PR DESCRIPTION
**Pull request by delputnam**
Depends on D9660-code.
 Testing:

Hack the back end to start a transfer with delayed transfer start state
Make sure the domains api returns `pending_start` for the domain.

Visit the manage domains page for the site.
Make sure the "Domain transfer waiting" notice is shown.
Make sure the "Action required" flag is shown.
Make sure the card with the "Start transfer" button is shown.
Make sure that "Start transfer" successfully calls the `inbound-transfer-start` endpoint.

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 1]
```diff
@@ -31,21 +31,22 @@ import InboundTransferEmailVerificationCard from 'my-sites/domains/domain-manage
 
 class Transfer extends React.PureComponent {
 	state = {
-		isRestartingTransfer: false,
+		isSumitting: false,
```
Comment (aidvu) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: :P

File: client/my-sites/domains/domain-management/components/domain/transfer-flag.jsx [hunk 2]
```diff
@@ -34,6 +34,9 @@ class DomainTransferFlag extends PureComponent {
 			case transferStatus.PENDING_OWNER:
 				message = translate( 'Email Confirmation Required' );
 				break;
+			case transferStatus.PENDING_START:
+				message = translate( 'Action Required' );
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/components/domain/transfer-flag.jsx]: Hi! I've found a possible matching string that has already been translated **45** times:
`translate( 'Action required!' )` *ES Score: 17*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 3]
```diff
@@ -94,9 +130,32 @@ class Transfer extends React.PureComponent {
...
+		startInboundTransfer( selectedSite.ID, domain.name, ( error, result ) => {
+			if ( result ) {
+				this.props.successNotice( translate( 'The transfer has been successfully started.' ), {
+					duration: 5000,
+				} );
+				fetchDomains( selectedSite.ID );
+				this.toggleSubmittingState();
+			} else {
+				this.props.errorNotice(
+					error.message || translate( 'We were unable to start the transfer.' ),
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'We were unable to restart the transfer.' )` *ES Score: 6*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 4]
```diff
@@ -63,10 +64,45 @@ class Transfer extends React.PureComponent {
...
+						</p>
+					</div>
+					<div>
+						<Button
+							className="edit__transfer-button-fail"
+							onClick={ this.startTransfer }
+							busy={ isSumitting }
+							disabled={ isSumitting }
+						>
+							{ isSumitting ? translate( 'Starting Transfer…' ) : translate( 'Start Transfer' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Start Transfer Again' )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/inbound-transfer-nux-start)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 5]
```diff
@@ -94,9 +130,32 @@ class Transfer extends React.PureComponent {
...
+	startTransfer = () => {
+		const { domain, selectedSite, translate } = this.props;
+		this.toggleSubmittingState();
+
+		startInboundTransfer( selectedSite.ID, domain.name, ( error, result ) => {
+			if ( result ) {
+				this.props.successNotice( translate( 'The transfer has been successfully started.' ), {
+					duration: 5000,
+				} );
+				fetchDomains( selectedSite.ID );
```
Comment (aidvu) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: Nice!

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 6]
```diff
@@ -94,9 +130,32 @@ class Transfer extends React.PureComponent {
 		this.props.paymentSettingsClick( this.props.domain );
 	};
 
+	startTransfer = () => {
```
Comment (aidvu) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: Cool! Didn't need this here, as we will move it to precheck, but I like it!

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 7]
```diff
@@ -63,10 +64,45 @@ class Transfer extends React.PureComponent {
...
+						</p>
+					</div>
+					<div>
+						<Button
+							className="edit__transfer-button-fail"
+							onClick={ this.startTransfer }
+							busy={ isSubmitting }
+							disabled={ isSubmitting }
+						>
+							{ isSubmitting ? translate( 'Starting Transfer…' ) : translate( 'Start Transfer' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: Hi! I've found a possible matching string that has already been translated **18** times:
`translate( 'Start Transfer Again' )` *ES Score: 13*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/inbound-transfer-nux-start)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/domains/components/domain-warnings/index.jsx [hunk 8]
```diff
@@ -843,6 +843,25 @@ export class DomainWarnings extends React.PureComponent {
 					}
 				);
 				break;
+			case transferStatus.PENDING_START:
+				compactMessage = translate( 'Domain transfer waiting' );
+				message = translate(
+					'Your domain {{strong}}%(domain)s{{/strong}} is witing for you to start the transfer. {{a}}More info{{/a}}',
```
Comment (aidvu) [inline client/my-sites/domains/components/domain-warnings/index.jsx]: `witing ` typo.

**Pull request merged**